software/24.Git_GitHub/Git/README.md

```diff
@@ -0,0 +1,189 @@
+To see those aliases it's simple: `alias | grep git`
+
+## Step 1 - Initialize a repository locally
+
+There are multiple ways to initialize a repository locally.
+
+### Clone an existing repository
+
+The most known one is to clone a repository using `git clone`.
+
+It is recommended to clone via ssh rather than https.
+SSH repository URLs can be identified as they start with `git@github.com:` whereas HTTPS URLs start with `https://github.com/`
+The reason why ssh is recommended is that you don't have to enter your username and password everytime you want to pull or push because your ssh key authenticates you, and it is more secured than your password.
+
+If you haven't already, go ahead and clone the GitHub repository created for this workshop.
+
+### Initialize a directory as a repository
+
+You can also simply initialize a directory as a git repository by using `git init`
+
+Once again it is recommended to use the SSH URL instead of the HTTPS one.
+
+_This part doesn't require you to do anything it's mainly for knowledge_
+
+## Step 2 - Modifying repository content
+
+### Control what files can be added
+
+It is recommended to use a `.gitignore` file to list all the files you never want to push (like binaries, or your local IDE configs).
+You can even have generated ones based on the language you're developing in.
+
+### Check what's been modified
+
+You have multiple ways of doing that.
+
+The first one is to check the status of your local repository with `git status`, you will see what file has been changed.
+
+After that, you can see exactly what lines have been modified using `git diff`.
+
+### Add Modifications to the index
+
+When you want to modify the repository content, you have multiple ways of doing that.
+
+The one that you might already know is using `git add`, a lot of people use `git add .` at the root of the repository, but it is not recommended because you might add files that you do not want to.
+Also, a better way of adding everything is using `git add --all`, because if you're not at the root of your repository, one will work and not the other.
+
+`git add .` is more intended to be used whe you're wandering around in your repo and want to add the folder you're in.
+
+You also have other commands that can be used to modify files such as `git rm` and `git mv`, they are respectively used to remove and move a file.
+
+Now add a file named `step2.txt` with "Hello World!" written in it.
+
+### Commit your modifications
+
+Once you have modified your files and added these modifications to the index, you can go ahead and commit those.
+Keep in mind that a single commit should make as few changes as possible, and should keep the changes on the same topic.
+It is completely fine to commit a single line.
+
+I recommend using `git commit` without the `-m` option, this will open your default editor and allow you to write a formatted message.
```

Comment:
The -m may be practical but people may tend to use it too often and create commits with too few explanations on what has been changed.
So I really think it will be beneficial for everyone to not use the -m.
Furthermore, it allows to quickly see the list of files changed, it allows you to double check if everything has been entered which has saved me a few times from forgetting to commit files.